web/src/message_fetch.js

```diff
@@ -49,6 +51,15 @@ function process_result(data, opts) {
 
     if (messages.length !== 0) {
         message_util.add_old_messages(messages, opts.msg_list);
+
+        // We keep track of how far back we've fetched messages for, for messaging in
+        // the recent view. This assumes `data.messages` is already sorted.
+        const oldest_timestamp = data.messages[messages.length - 1].timestamp;
```

Comment:
This calculation is also wrong, in that we have no guarantee whether the new batch of messages is at older or new. 

I think instead you want to query `all_messages_data.last()`. Also need to handle the case that there are no messages and that is undefined.

Reply:
> you want to query all_messages_data.last()

Ah cool, switched to that

> Also need to handle the case that there are no messages and that is undefined.

I believe this is already handled by this being in an `if (messages.length !== 0)` -- unless they're in a fresh org with no messages at all, eventually messages will load which will trigger the banner to update.

Reply:
Well, we don't want this to crash of give wrong results if you're like a guest user who doesn't have access to any messages.

Reply:
But I think that's fine because `all_messages_data`, by definition, does no filtering, so stuff like muting can't result in there being actually 0 messages there.